projects/detector/private/Path.cxx

```diff
@@ -249,14 +261,16 @@ void Path::ClipToOuterBounds() {
         if(dot < 0) {
             p0.swap(p1);
         }
-        bool clip_0 = (p0 - first_point_) * direction_ > 0;
-        bool clip_1 = (p1 - last_point_) * direction_ < 0;
+        bool clip_0 = ((p0 - first_point_) * direction_ > 0) || (first_inf_);
```

Comment:
It might make more sense to swap the order of the two statements in the `||` operator so that `first_inf_` prevents the latter calculation from being performed if it is True.